ДЗ спринт №8

morechiba opened the pull request.

morechiba committed:
верстка двух экранов - главный и настройки

morechiba committed:
релизация клика на кнопки с эффектом:
1) анонимный класс
2 и 3) лямбда-выражения

morechiba committed:
новые экраны - медиа и поиск
на главном экране реализован переход на новые экраны (заглушки убраны)

kekulta commented on app/src/main/java/com/tagomago/playlistmaker/MainActivity.kt:
```diff
@@ -1,20 +1,46 @@
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        enableEdgeToEdge()
+
         setContentView(R.layout.activity_main)
-        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
-            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
-            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
-            insets
```
Очень рекомендую разобраться edgeToEdge. Гугл включает его по умолчанию в новых Андроидах.

kekulta commented on app/src/main/java/com/tagomago/playlistmaker/MediaLibraryActivity.kt:
```diff
@@ -0,0 +1,22 @@
+        setContentView(R.layout.activity_media_library)
+
+        val backButton = findViewById<Button>(R.id.back)
+        backButton.setOnClickListener {
+            val displayIntent = Intent(this, MainActivity::class.java)
+            startActivity(displayIntent)
+        }
```
Так делать неправильно. Когда ты создаёшь новую активити, то она добавляется в стек поверх старой.
Если ты будешь открывать и закрывать этот экран, то ты будешь поочерёдно класть в стек главный и этот экраны.
Стек будет расти и в какой-то момент это может привести к проблемам с памятью.
Правильный подход закрывать активити, когда уходишь с экран, например, с помощью метод `finish()`.

kekulta commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,130 @@
+            android:paddingRight="2dp"
+            android:paddingTop="3dp"/>
+    </FrameLayout>
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/list_height"
+        android:paddingHorizontal="@dimen/padding_main"
+        android:paddingTop="@dimen/list_padding"
+        >
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textSize="@dimen/text_small"
+            android:text="@string/privacy"
+            android:textColor="@color/black"
+            android:layout_gravity="start"
+            android:paddingTop="@dimen/list_padding"
+            />
+        <ImageView
+            android:layout_width="24dp"
+            android:layout_height="24dp"
+            android:src="@drawable/arr"
+            android:layout_gravity="end"
+            android:layout_marginTop="8dp"
+            android:layout_marginRight="2dp"
+            android:paddingRight="3dp"
+            android:paddingTop="4dp"/>
```
Можно заменить на Compound Drawable.

morechiba committed:
цвет иконок вынесен в ресурсы

morechiba committed:
оптимизация верстки

morechiba committed:
вернула enableEdgeToEdge() в Activity (Гугл включает его по умолчанию в новых Андроидах)

morechiba committed:
1) вторая часть enableEdgeToEdge() для Activity - отступ для статус-бара и пр.
2) немного скорректировала верстку для правильного отображения activity_main при добавке edge-to-edge
3) исправлено переключение между экранами - finish() для активного экрана, чтобы убрать его из стека (иначе будет множиться и жрать память)

morechiba committed:
для главного экрана тоже добавлено закрытие Activity при переключении на другие экраны

morechiba committed:
стили перенесены в ресурсный файл
добавлены шрифты

morechiba committed:
вылезла ошибка, пока непонятно
старт страницы settings

morechiba committed:
исправлены ошибки в стилях
+убрана тень у кнопок

morechiba committed:
начало действий на странице настроек +в строковые ресурсы вынесен текст+email

morechiba committed:
перевод текста для шаринга

morechiba committed:
доделана страница настроек

morechiba committed:
верстка строки поиска

morechiba committed:
clear_button в строке поиска
правильный цвет у textColorHint

morechiba committed:
убирание клавиатуры после очистки поля ввода
